File: sources/iOS/ios-communications/Sources/PolarBleSdk/sdk/api/model/activity/PolarActiveTimeData.swift
///  Copyright © 2024 Polar. All rights reserved.

import Foundation

public struct PolarActiveTimeData: Codable  {
    public let date: Date
    public let timeNonWear: PolarActiveTime
    public let timeSleep: PolarActiveTime
    public let timeSedentary: PolarActiveTime
    public let timeLightActivity: PolarActiveTime
    public let timeContinuousModerateActivity: PolarActiveTime
    public let timeIntermittentModerateActivity: PolarActiveTime
    public let timeContinuousVigorousActivity: PolarActiveTime
    public let timeIntermittentVigorousActivity: PolarActiveTime

    public init(date: Date,
                timeNonWear: PolarActiveTime = PolarActiveTime(),
                timeSleep: PolarActiveTime = PolarActiveTime(),
                timeSedentary: PolarActiveTime = PolarActiveTime(),
                timeLightActivity: PolarActiveTime = PolarActiveTime(),
                timeContinuousModerateActivity: PolarActiveTime = PolarActiveTime(),
                timeIntermittentModerateActivity: PolarActiveTime = PolarActiveTime(),
                timeContinuousVigorousActivity: PolarActiveTime = PolarActiveTime(),
                timeIntermittentVigorousActivity: PolarActiveTime = PolarActiveTime()) {
        self.date = date
        self.timeNonWear = timeNonWear
        self.timeSleep = timeSleep
        self.timeSedentary = timeSedentary
        self.timeLightActivity = timeLightActivity
        self.timeContinuousModerateActivity = timeContinuousModerateActivity
        self.timeIntermittentModerateActivity = timeIntermittentModerateActivity
        self.timeContinuousVigorousActivity = timeContinuousVigorousActivity
        self.timeIntermittentVigorousActivity = timeIntermittentVigorousActivity
    }
}

public struct PolarActiveTime: Codable {
    public let hours: Int
    public let minutes: Int
    public let seconds: Int
    public let millis: Int

    public init(hours: Int = 0, minutes: Int = 0, seconds: Int = 0, millis: Int = 0) {
        self.hours = hours
        self.minutes = minutes
        self.seconds = seconds
        self.millis = millis
    }

    public func encode(to encoder: Encoder) throws {
        var container = encoder.singleValueContainer()
        try container.encode(hours*3600 + minutes*60 + seconds + millis/1000)
    }
}

File: sources/Android/android-communications/library/src/sdk/java/com/polar/sdk/api/model/activity/PolarActiveTimeData.kt
package com.polar.sdk.api.model.activity

import java.time.LocalDate

data class PolarActiveTimeData(
    val date: LocalDate,
    val timeNonWear: PolarActiveTime = PolarActiveTime(),
    val timeSleep: PolarActiveTime = PolarActiveTime(),
    val timeSedentary: PolarActiveTime = PolarActiveTime(),
    val timeLightActivity: PolarActiveTime = PolarActiveTime(),
    val timeContinuousModerateActivity: PolarActiveTime = PolarActiveTime(),
    val timeIntermittentModerateActivity: PolarActiveTime = PolarActiveTime(),
    val timeContinuousVigorousActivity: PolarActiveTime = PolarActiveTime(),
    val timeIntermittentVigorousActivity: PolarActiveTime = PolarActiveTime()
)

data class PolarActiveTime(
    val hours: Int = 0,
    val minutes: Int = 0,
    val seconds: Int = 0,
    val millis: Int = 0
)